### ISSUE TITLE
What does this error mean?

### ISSUE THREAD

**Issue by premier69** [Genymobile/scrcpy#3259]
This used to work perfectly with my dead old xiomi mi max 3 but my new redmi note 10 gives me this:

```
INFO: scrcpy 1.12.1 <https://github.com/Genymobile/scrcpy>
adb server version (41) doesn't match this client (39); killing...
* daemon started successfully
/usr/share/scrcpy/scrcpy-server: 1 fil...shed. 0.9 MB/s (24773 bytes in 0.025s)
INFO: Initial texture: 1080x2400
WARN: Device disconnected
[server] ERROR: Exception on thread Thread[main,5,main]
java.lang.AssertionError: java.lang.reflect.InvocationTargetException
	at com.genymobile.scrcpy.wrappers.SurfaceControl.setDisplaySurface(SurfaceControl.java:75)
	at com.genymobile.scrcpy.ScreenEncoder.setDisplaySurface(ScreenEncoder.java:178)
	at com.genymobile.scrcpy.ScreenEncoder.streamScreen(ScreenEncoder.java:70)
	at com.genymobile.scrcpy.Server.scrcpy(Server.java:34)
	at com.genymobile.scrcpy.Server.main(Server.java:163)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:357)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.genymobile.scrcpy.wrappers.SurfaceControl.setDisplaySurface(SurfaceControl.java:73)
```

**Comment by rom1v** [comment]
It means yoj use Android 12, which requires scrcpy >= 1.18, but you use an older version. #2129

I guess you are using Ubuntu 20.04 or similar. Uninstall the scrcpy package and install the latest version manually: https://github.com/Genymobile/scrcpy/blob/master/BUILD.md#simple

**Comment by premier69** [comment]
Thank You! and yes I'm using linux mint. i uninstalled scrcpy and scrcpy-server with the gui software manager but left guiscrcpy

i installed per the instructions:

# for Debian/Ubuntu
sudo apt install ffmpeg libsdl2-2.0-0 adb wget \
                 gcc git pkg-config meson ninja-build libsdl2-dev \
                 libavcodec-dev libavdevice-dev libavformat-dev libavutil-dev \
                 libusb-1.0-0 libusb-1.0-0-dev
git pull
./install_release.sh

i still have the same problem with the terminal giving me this output when i enter the scrcpy command:

scrcpy 1.24 <https://github.com/Genymobile/scrcpy>
/usr/local/share/scrcpy/scrcpy-server:...shed. 6.1 MB/s (41159 bytes in 0.006s)
[server] INFO: Device: Xiaomi M2101K6G (Android 12)
INFO: Renderer: opengl
INFO: OpenGL version: 4.6 (Compatibility Profile) Mesa 22.0.3 - kisak-mesa PPA
INFO: Trilinear filtering enabled
INFO: Initial texture: 1080x2400
[server] ERROR: Encoding error: java.lang.IllegalStateException: null
[server] ERROR: Exception on thread Thread[main,5,main]
java.lang.IllegalStateException
	at android.media.MediaCodec.native_dequeueOutputBuffer(Native Method)
	at android.media.MediaCodec.dequeueOutputBuffer(MediaCodec.java:3535)
	at com.genymobile.scrcpy.ScreenEncoder.encode(ScreenEncoder.java:152)
	at com.genymobile.scrcpy.ScreenEncoder.internalStreamScreen(ScreenEncoder.java:102)
	at com.genymobile.scrcpy.ScreenEncoder.streamScreen(ScreenEncoder.java:74)WARN: Device disconnected

	at com.genymobile.scrcpy.Server.scrcpy(Server.java:101)
	at com.genymobile.scrcpy.Server.main(Server.java:335)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:357)

**Comment by rom1v** [comment]
> i still have the same problem

No, it's not the same, now your error is a bug in MIUI 13, see #3051.

**Comment by icy-comet** [comment]
Shouldn't this be also closed @rom1v in favor of #3051 ?
